src/cred.c

```diff
@@ -736,6 +736,88 @@ fido_cred_set_attstmt(fido_cred_t *cred, const unsigned char *ptr, size_t len)
 	return (r);
 }
 
+static int
+parse_attobj(const cbor_item_t *key, const cbor_item_t *val, void *arg)
```

Comment:
We have a very similar function in `winhello.c`,  see `decode_attobj()`.  If we were to introduce this public API, I feel like it'd make sense to consolidate them.  

One approach would be to move `winhello.c`'s `decode_attobj()` to e.g. `cbor.c` and introduce a `cbor_decode_attobj()` function, like you've done (but I don't think we need it to take a function pointer). Then call this function from both `winhello.c/translate_winhello_cred()` and the new `fido_cred_set_attobj()`.

To ensure a consistent state, `fido_cred_set_attobj()` would have to call `fido_cred_reset_rx()` first.

Any thoughts?

Reply:
Did not know that winhello had same function, so ended up adding a new one; fixed now.
`fido_cred_reset_rx` was tempting but it clears large-blob too and I don't know what it does. Anyways, now using `fido_cred_reset_rx`.
Pushed new changes according to your suggestion.

Reply:
Ok, in that case I will add `fido_cred_clean_attobj()`. 
Should `winhello.c/translate_winhello_cred()` also be updated to use it? @LDVG

Reply:
That should not be necessary, `fido_winhello_make_cred()` already calls `fido_cred_reset_rx()` before calling `translate_winhello_cred()`

Reply:
Ok, pushed a new change to add `fido_cred_clean_attobj()`.